python/examples/extract_position_data.py

```diff
@@ -68,26 +70,61 @@
+    if len(pose_data.messages) == 0:
+        logger.warning('No pose data found in log file.')
+        sys.exit(2)
 
     # Generate a CSV file.
-    pose_data = result[PoseMessage.MESSAGE_TYPE]
-    with open(os.path.join(output_dir, 'position.csv'), 'w') as f:
-        f.write('GPS Time (sec), Solution Type, Lat (deg), Lon (deg), Ellipsoid Alt (m)\n')
-        for message in pose_data.messages:
-            if message.solution_type != SolutionType.Invalid and message.gps_time:
-                f.write('%.3f, %d, %.8f, %.8f, %.3f\n' %
-                        (float(message.gps_time), message.solution_type.value, *message.lla_deg))
+    path = os.path.join(output_dir, 'position.csv')
+    logger.info("Generating '%s'." % path)
+    with open(path, 'w') as f:
+        f.write('P1 Time (sec), GPS Time (sec), Solution Type, Lat (deg), Lon (deg), Ellipsoid Alt (m), # Satellites\n')
+
+        # If we do not have satellite data present in the log, we'll simply loop over the pose messages and write them
+        # to the file.
+        if len(satellite_data.messages) == 0:
+            for message in pose_data.messages:
+                if message.solution_type != SolutionType.Invalid:
+                    # Note we set # satellites to 0 here since it is unknown.
+                    f.write('%.6f, %.6f, %d, %.8f, %.8f, %.3f, %d\n' %
+                            (float(message.p1_time), float(message.gps_time),
+                             message.solution_type.value, *message.lla_deg, 0))
+        # If we do have satellite data, we can then time-align the two message types using their P1 timestamps. That
+        # way, the generated CSV file can include both position and satellite count for each time.
+        else:
+            # First, find the intersection of the pose and satellite data. Note that we do not assume we have the same
+            # number of messages for both types since, if the data was recorded remotely, we may have connected to the
+            # device between messages.
+            _, pose_idx, satellite_idx = np.intersect1d(pose_data.p1_time, satellite_data.p1_time, return_indices=True)
+
+            # Generate a set of satellite counts for each position epoch.
+            num_svs = np.full_like(pose_data.p1_time, 0, dtype=int)
+            num_svs[pose_idx] = satellite_data.num_svs[satellite_idx]
+
+            # Write out the position data with corresponding SV counts, or 0 where not available.
+            for i in range(len(pose_data.p1_time)):
+                f.write('%.6f, %.6f, %d, %.8f, %.8f, %.3f, %d\n' %
+                        (pose_data.p1_time[i], pose_data.gps_time[i],
+                         pose_data.solution_type[i], *pose_data.lla_deg[:, i], num_svs[i]))
 
     # Generate a KML file.
-    with open(os.path.join(output_dir, 'position.kml'), 'w') as f:
+    path = os.path.join(output_dir, 'position.kml')
+    logger.info("Generating '%s'." % path)
+    with open(path, 'w') as f:
```

Comment:
We should add timestamps to our KML

Reply:
I'm not super familiar with how to do that. We're using a list of coordinates inside of a line string:
```
<LineString>
  <coordinates>
    -127.456,37.123
    -127.45678,37.12345
  </coordinates>
</LineString>
```
I don't think a `<coordinates>` element can be assigned timestamps for individual positions. I think you can set a `gx:TimeSpan` property on the `<LineString>` but that would just be the total range.

Reply:
Here's a gist showing an example, but I think we can do this in a separate PR: https://gist.github.com/jsmorph/26c63d2bf84b7e2e041f